elemica-suggest.coffee

```diff
@@ -154,10 +160,11 @@ elemicaSuggest 0.7.1-SNAPSHOT.
         $(this).attr 'autocomplete', 'off'
 
         $(this).on 'blur', (event) ->
-          removeSuggestions event.target
+          $target = $(event.target)
+          removeSuggestions $target
 
           if $valueInput.val() == ""
-            $(event.target).val("")
+            $target.val("") if noSuggestionMatched($target.val())
             afterSelect(null)
```

Comment:
What if user wants to clear typeahead input and run code in `afterSelect`? 

`afterSelect` accepts selected value as parameter, so user can decide whether to dynamically add values or not.